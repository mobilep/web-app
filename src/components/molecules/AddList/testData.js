export const users = [
	{
		id: 1,
		fullName: 'John Smith',
		initials: 'JS',
	}, {
		id: 2,
		fullName: 'Philippe Starck',
		initials: 'PS',
	}, {
		id: 3,
		fullName: 'Suzy Salmon',
		initials: 'SS',
	},
];

export const allUsers = [
	{
		id: 4,
		fullName: 'John Rebele',
		initials: 'JR',
	}, {
		id: 5,
		fullName: 'John Klokdan',
		initials: 'JK',
	}, {
		id: 6,
		fullName: 'John Makeba',
		initials: 'JM',
	},
];
