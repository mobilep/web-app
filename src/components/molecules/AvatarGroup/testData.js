export const avatars = [
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
];

export const avatarsLess = [
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
];
