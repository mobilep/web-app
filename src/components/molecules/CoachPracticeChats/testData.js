/* eslint-disable max-len */
export const practiceChats = [
	{
		currentUserId: '5d6e509093b6e912b83a65ce',
		_id: '5dad8ef5d3c420b10c618100',
		_coach: {
			_id: '5d6e509093b6e912b83a65ce',
			firstName: 'Roman',
			lastName: 'Yavoriv',
			avatarId: '5f549515-ee86-40f2-9104-5c6a8a80beb7',
			devices: 'Array[0]',
			avatarColor: '58c9ef',
			lang: 'en',
			avatar_sm: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/100/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			avatar_md: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/640/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			avatar_lg: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/1024/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			havePassword: false,
			name: 'Roman Yavoriv',
			extraInformation: 'Array[7]',
		},
		_scenario: {
			_id: '5dad8ef546396073dd363907',
			groupChat: 'https://mobile-practice-dev.firebaseio.com/mpchat/5dad8ef546396073dd363909',
			isPracticed: true,
			isActive: true,
			name: '2nd scenario',
			video: 'Object',
		},
		_user: {
			_id: '5d70bc6d93b6e912b83a65d5',
			firstName: 'Taras',
			lastName: 'Kvas',
			avatarId: '2e3db607-9ec1-45f4-a7e5-4bed03912dd5',
			devices: 'Array[0]',
			avatarColor: 'fd524f',
			lang: 'en',
			avatar_sm: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/100/2e3db607-9ec1-45f4-a7e5-4bed03912dd5.jpg',
			avatar_md: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/640/2e3db607-9ec1-45f4-a7e5-4bed03912dd5.jpg',
			avatar_lg: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/1024/2e3db607-9ec1-45f4-a7e5-4bed03912dd5.jpg',
			havePassword: false,
			name: 'Taras Kvas',
			extraInformation: 'Array[7]',
		},
		__v: 0,
		createdAt: '2019-10-21T10:56:53.554Z',
		updatedAt: '2019-10-29T09:19:42.764Z',
		userEvaluatedAt: null,
		coachEvaluatedAt: null,
		hasBestPractice: false,
		hasUserMessage: true,
		isActive: true,
		lastMessageFrom: 'learner',
		status: 'current',
		coachMark: [],
		userMark: [],
		coachAudiosCount: 0,
		coachPhotosCount: 0,
		coachFilesCount: 0,
		coachVideosCount: 0,
		coachMessagesCount: 4,
		userAudiosCount: 0,
		userPhotosCount: 1,
		userFilesCount: 0,
		userVideosCount: 0,
		userMessagesCount: 19,
		_inbox: {
			_id: '5dad8ef546396073dd36390c',
			updatedAt: '2019-10-29T09:20:28.648Z',
			createdAt: '2019-10-21T10:56:53.544Z',
			_user: '5d6e509093b6e912b83a65ce',
			_recipient: '5d70bc6d93b6e912b83a65d5',
			__v: 0,
			inboxId: 'chats/5d6e509093b6e912b83a65ce/5dad8ef546396073dd363907/5dad8ef5d3c420b10c618100/5dad8ef546396073dd36390c',
			lastReadMessage: '-LsLo4OhHveQpLP5JXKy',
			unreadMessagesCount: 0,
			type: 'practice',
			status: 'read',
			message: 'Object',
			firstResponseAt: '2019-10-28T13:25:34.142Z',
			firstMessageAt: '2019-10-28T13:09:17.159Z',
			isActive: true,
		},
		state: 'waitingOnFeedback',
		messages: [
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
		],
	},
	{
		currentUserId: '5d6e509093b6e912b83a65ce',
		_id: '5dad8ef5d3c420b10c618102',
		_coach: {
			_id: '5d6e509093b6e912b83a65ce',
			firstName: 'Roman',
			lastName: 'Yavoriv',
			avatarId: '5f549515-ee86-40f2-9104-5c6a8a80beb7',
			devices: 'Array[0]',
			avatarColor: '58c9ef',
			lang: 'en',
			avatar_sm: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/100/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			avatar_md: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/640/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			avatar_lg: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/1024/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			havePassword: false,
			name: 'Roman Yavoriv',
			extraInformation: 'Array[7]',
		},
		_scenario: {
			_id: '5dad8ef546396073dd363907',
			groupChat: 'https://mobile-practice-dev.firebaseio.com/mpchat/5dad8ef546396073dd363909',
			isPracticed: true,
			isActive: true,
			name: '2nd scenario',
			video: 'Object',
		},
		_user: {
			_id: '5da86a3716973b29347321ce',
			firstName: 'Roman',
			lastName: 'Yavoriv',
			devices: 'Array[0]',
			avatarColor: '00cdbb',
			lang: 'en',
			avatar_sm: null,
			avatar_md: null,
			avatar_lg: null,
			havePassword: false,
			name: 'Roman Yavoriv',
			extraInformation: 'Array[7]',
		},
		__v: 0,
		createdAt: '2019-10-21T10:56:53.563Z',
		updatedAt: '2019-10-29T09:18:36.340Z',
		userEvaluatedAt: '2019-10-29T09:16:14.949Z',
		coachEvaluatedAt: null,
		hasBestPractice: false,
		hasUserMessage: false,
		isActive: true,
		lastMessageFrom: 'coach',
		coachMark: [
			'Object',
		],
		userMark: [],
		coachAudiosCount: 0,
		coachPhotosCount: 0,
		coachFilesCount: 0,
		coachVideosCount: 0,
		coachMessagesCount: 2,
		userAudiosCount: 0,
		userPhotosCount: 0,
		userFilesCount: 0,
		userVideosCount: 0,
		userMessagesCount: 0,
		_inbox: {
			_id: '5dad8ef546396073dd36390d',
			updatedAt: '2019-10-29T09:19:11.848Z',
			createdAt: '2019-10-21T10:56:53.547Z',
			_user: '5d6e509093b6e912b83a65ce',
			_recipient: '5da86a3716973b29347321ce',
			__v: 0,
			inboxId: 'chats/5d6e509093b6e912b83a65ce/5dad8ef546396073dd363907/5dad8ef5d3c420b10c618102/5dad8ef546396073dd36390d',
			lastReadMessage: '-LsLnpAfH4s1IOZsQrgY',
			unreadMessagesCount: 0,
			type: 'practice',
			status: 'read',
			message: 'Object',
			firstResponseAt: null,
			firstMessageAt: '2019-10-29T09:18:27.588Z',
			isActive: true,
		},
		state: 'completed',
		messages: [
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
		],
	},
];

export const activeChatId = practiceChats[0]._id;
