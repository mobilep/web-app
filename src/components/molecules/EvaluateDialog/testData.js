export const evaluationList = [
	{
		id: '1',
		title: 'Contactâ€™s attention grabbed',
	}, {
		id: '2',
		title: 'Self Introduction',
	}, {
		id: '3',
		title: 'Purpose of call stated',
	},
];
