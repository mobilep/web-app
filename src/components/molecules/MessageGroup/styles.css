:root {
	--MessageGroup-avatarSize: 50px;
	--MessageGroup-messagesMargin: 0 20px;
}

.MessageGroup {
	display: flex;
	margin-top: 20px;
	align-self: flex-start;
}

.MessageGroup:not(.incoming) {
	align-self: flex-end;
	flex-direction: row-reverse;
}

.MessageGroup .Avatar {
	min-width: var(--MessageGroup-avatarSize);
	width: var(--MessageGroup-avatarSize);
	height: var(--MessageGroup-avatarSize);
	font-size: calc(var(--MessageGroup-avatarSize) / 2.5);
}

.MessageGroup-content {
	display: flex;
	flex-direction: column;
	margin: var(--MessageGroup-messagesMargin);
	max-width: 100%;
	width: 100%;
}

.MessageGroup-messages {
	display: flex;
	flex-direction: column;
}

.MessageGroup-message,
.MessageGroup-message > div {
	max-width: 100%;
}

.MessageGroup:not(.incoming) .MessageGroup-messages {
	align-items: flex-end;
}

.MessageGroup-message:not(:first-child) {
	margin-top: 10px;
}

@media (max-width: 768px) {
	.MessageGroup .Avatar {
		min-width: 28px;
		width: 28px;
		height: 28px;
		font-size: calc(28px / 2.5);
	}

	.MessageGroup-content {
		display: flex;
		flex-direction: column;
		margin: 0 5px;
	}

	.MessageGroup-message {
		max-width: 95%;
	}
}
