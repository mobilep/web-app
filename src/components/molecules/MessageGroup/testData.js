/* eslint-disable max-len */
export default {
	author: {
		_company: '592b3b608cd6291923e74ac1',
		_id: '592b3b608cd6291923e74ac2',
		isCompanyAdmin: true,
		isSysAdmin: false,
		email: 'mike@company.com',
		lastName: 'Mike',
		firstName: 'King',
	},
	messages: [
		{
			type: 'text',
			user: '592b3b608cd6291923e74ac2',
			content: 'Here’s my first good attempt. Let me know what you think.Good start. Please re-record, remember to slow down at the beginning and focus on customer needs even more',
			time: '1525653351987',
			messageId: 'key11',
		},
		{
			type: 'video',
			user: '592b3b608cd6291923e74ac2',
			content: {
				id: '123',
				thumbnailUrl: 'http://via.placeholder.com/350x150/00423b?text=video%20message',
			},
			time: '1525653371987',
			messageId: 'key12',
		},
		{
			type: 'text',
			user: '592b3b608cd6291923e74ac2',
			content: 'Here’s my first good attempt. Let me know what you think.',
			time: '1525675351987',
			messageId: 'key13',
		},
		{
			type: 'evaluation',
			user: '592b3b608cd6291923e74ac2',
			content: {
				avgMark: '4.3',
				text: 'Here’s my first good attempt. Let me know what you think, please !',
			},
			time: '1525854790993',
			messageId: 'key14',
		},
	],
	time: 1525653371987,
};
