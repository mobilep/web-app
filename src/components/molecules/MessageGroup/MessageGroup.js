import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';

import { Avatar } from '../..';
import Message from '../Message';

import { getInitials } from '../../../helpers';
import { getTimeString } from '../../../utils/time';

import './styles.css';

class MessageGroup extends Component {

	static propTypes = {
		author: PropTypes.object,
		children: PropTypes.any,
		className: PropTypes.string,
		evaluateText: PropTypes.string,
		incoming: PropTypes.bool,
		justSentText: PropTypes.string,
		last: PropTypes.bool,
		lastReadMessage: PropTypes.string,
		messages: PropTypes.array,
		onContextMenu: PropTypes.func,
		onEvaluateClick: PropTypes.func,
		onViewDetailsClick: PropTypes.func,
		scenarioId: PropTypes.string,
		scenarioOwner: PropTypes.bool,
		time: PropTypes.number,
		viewDetailsText: PropTypes.string,
		viewScenarioDetails: PropTypes.string,
		viewScenarioDetailsText: PropTypes.string,
	}

	static defaultProps = {
		author: {},
		messages: [],
		onContextMenu: () => {},
	}

	getClassName () {
		const { className, incoming } = this.props;

		return classNames({ incoming }, 'MessageGroup', className);
	}

	getMessagesView () {
		const { messages } = this.props;

		return messages.map(this.getMessageView);
	}

	getMessageView = (message, index) => {
		const {
			incoming,
			onContextMenu,
			viewDetailsText,
			viewScenarioDetailsText,
			evaluateText,
			onEvaluateClick,
			onViewDetailsClick,
			scenarioOwner,
			scenarioId,
			messages,
			justSentText,
			author,
			lastReadMessage,
			children,
		} = this.props;

		const last = index === messages.length - 1;
		const timeString = getTimeString({ justSentText, time: +message.time, last, author: author.firstName });

		return (
			<Fragment key={`${message.messageId}`}>
				<Message
					className='MessageGroup-message'
					message={{ ...message, timeString }}
					incoming={incoming}
					singleOrLast={last}
					onContextMenu={onContextMenu}
					viewDetailsText={viewDetailsText}
					viewScenarioDetailsText={viewScenarioDetailsText}
					evaluateText={evaluateText}
					onEvaluateClick={onEvaluateClick}
					onViewDetailsClick={onViewDetailsClick}
					scenarioOwner={scenarioOwner}
					scenarioId={scenarioId}
				/>
				{ message.messageId === lastReadMessage && !last && children }
			</Fragment>
		);
	}

	render () {
		const {
			author,
			messages,
		} = this.props;
		if (messages.length === 0) return null;

		const initials = getInitials(author.firstName, author.lastName);
		const messagesView = this.getMessagesView();

		return (
			<div className={this.getClassName()}>
				<Avatar
					size={Avatar.AVATAR_SIZES.AVATAR_28}
					initials={initials}
					color={`#${author.avatarColor}`}
					img={author.avatar_md}
				/>
				<div className='MessageGroup-content'>
					<div className='MessageGroup-messages'>{messagesView}</div>
				</div>
			</div>
		);
	}
}

export default MessageGroup;
