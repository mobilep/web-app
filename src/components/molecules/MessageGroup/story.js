import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { boolean, number } from '@storybook/addon-knobs/react';
import MessageGroup from './MessageGroup';
import testData from './testData';

storiesOf('2. Molecules / MessageGroup', module)
	.add('default', () => (
		<MessageGroup
			author={testData.author}
			incoming={boolean('Incoming', true)}
			last={boolean('Last', true)}
			messages={testData.messages}
			time={number('Time', testData.time)}
			onContextMenu={action('onContextMenu')}
			onVideoMessageCancelUpload={action('onVideoMessageCancelUpload')}
			onVideoMessagePlay={action('onVideoMessagePlay')}
		/>
	));
