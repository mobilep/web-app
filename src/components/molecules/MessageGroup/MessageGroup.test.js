import React from 'react';
import { shallow } from 'enzyme';

import MessageGroup from './MessageGroup';
import testData from './testData';

describe('Molecule - MessageGroup', () => {

	it('Should render', () => {
		const wrapper = shallow(<MessageGroup {...testData} />);
		expect(wrapper).toMatchSnapshot();
	});

});
