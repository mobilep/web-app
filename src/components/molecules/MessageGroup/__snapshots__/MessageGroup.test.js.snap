// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecule - MessageGroup Should render 1`] = `
<div
  className="MessageGroup"
>
  <Avatar
    border={false}
    color="#undefined"
    initials="KM"
    size="avatar-28"
  />
  <div
    className="MessageGroup-content"
  >
    <div
      className="MessageGroup-messages"
    >
      <Message
        className="MessageGroup-message"
        message={
          Object {
            "content": "Here’s my first good attempt. Let me know what you think.Good start. Please re-record, remember to slow down at the beginning and focus on customer needs even more",
            "messageId": "key11",
            "time": "1525653351987",
            "timeString": "King, 00:35",
            "type": "text",
            "user": "592b3b608cd6291923e74ac2",
          }
        }
        onContextMenu={[Function]}
        onVideoMessageCancelUpload={[Function]}
        onVideoMessagePlay={[Function]}
        singleOrLast={false}
        viewDetailsText="View details"
      />
      <Message
        className="MessageGroup-message"
        message={
          Object {
            "content": Object {
              "id": "123",
              "thumbnailUrl": "http://via.placeholder.com/350x150/00423b?text=video%20message",
            },
            "messageId": "key12",
            "time": "1525653371987",
            "timeString": "King, 00:36",
            "type": "video",
            "user": "592b3b608cd6291923e74ac2",
          }
        }
        onContextMenu={[Function]}
        onVideoMessageCancelUpload={[Function]}
        onVideoMessagePlay={[Function]}
        singleOrLast={false}
        viewDetailsText="View details"
      />
      <Message
        className="MessageGroup-message"
        message={
          Object {
            "content": "Here’s my first good attempt. Let me know what you think.",
            "messageId": "key13",
            "time": "1525675351987",
            "timeString": "King, 06:42",
            "type": "text",
            "user": "592b3b608cd6291923e74ac2",
          }
        }
        onContextMenu={[Function]}
        onVideoMessageCancelUpload={[Function]}
        onVideoMessagePlay={[Function]}
        singleOrLast={false}
        viewDetailsText="View details"
      />
      <Message
        className="MessageGroup-message"
        message={
          Object {
            "content": Object {
              "avgMark": "4.3",
              "text": "Here’s my first good attempt. Let me know what you think, please !",
            },
            "messageId": "key14",
            "time": "1525854790993",
            "timeString": "King, 08:33",
            "type": "evaluation",
            "user": "592b3b608cd6291923e74ac2",
          }
        }
        onContextMenu={[Function]}
        onVideoMessageCancelUpload={[Function]}
        onVideoMessagePlay={[Function]}
        singleOrLast={true}
        viewDetailsText="View details"
      />
    </div>
  </div>
</div>
`;
